test/libsolidity/SMTCheckerJSONTest.cpp

```diff
@@ -59,42 +62,59 @@ bool SMTCheckerTest::run(ostream& _stream, string const& _linePrefix, bool const
 
 	// This is the list of responses provided in the test
 	string auxInput("auxiliaryInput");
-	BOOST_CHECK(m_smtResponses.isMember(auxInput));
+	if (!m_smtResponses.isMember(auxInput))
+		BOOST_THROW_EXCEPTION(runtime_error("JSON file does not contain field \"auxiliaryInput\"."));
+
 	vector<string> inHashes = hashesFromJson(m_smtResponses, auxInput, "smtlib2responses");
 
 	// Ensure that the provided list matches the requested one
-	BOOST_CHECK_MESSAGE(
-		outHashes == inHashes,
-		"SMT query hashes differ: " + boost::algorithm::join(outHashes, ", ") + " x " + boost::algorithm::join(inHashes, ", ")
-	);
+	if (outHashes != inHashes)
+		BOOST_THROW_EXCEPTION(runtime_error(
+			"SMT query hashes differ: " +
+			boost::algorithm::join(outHashes, ", ") +
+			" x " +
+			boost::algorithm::join(inHashes, ", ")
+		));
 
 	// Rerun the compiler with the provided hashed (2nd run)
 	input[auxInput] = m_smtResponses[auxInput];
 	Json::Value endResult = compiler.compile(input);
 
-	BOOST_CHECK(endResult.isMember("errors"));
```

Comment:
Would be good to still have exceptions here.

Reply:
I actually put an `if` instead, such that it doesn't crash on cases that don't lead to assertion fails.

Reply:
Ah, I see!